### ISSUE TITLE
EXPath File Module: temporary files: $prefix and $suffix

### ISSUE THREAD

**Issue by michaelhkay** [expath/expath-cg#136]
The specifications for file:create-temp-file and file:create-temp-dir have arguments $prefix and $suffix, but the specifications says nothing about them. It appears to be the intent that the temporary files/directories that are created should have names beginning and ending with the supplied $prefix and $suffix. But the tests don't actually seem to require that.

**Comment by fgeorges** [comment]
Yes, it seems that way.  @ChristianGruen, it seems they were introduced in c61e21d5097bec89f89e378e44dd30b3192b237a, so I assign it to you.

**Comment by ChristianGruen** [comment]
Right, the spec is too inaccurate here. The prefix/suffix parameters have been a remnant of an early stage of the spec before I jumped in. I’m not sure if anyone requires them (well, too late).

**Comment by adamretter** [comment]
Could I add that the `$prefix` and `$suffix` look very similar to those arguments which are available in the JDK NIO for creating temp files, see: https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createTempFile-java.nio.file.Path-java.lang.String-java.lang.String-java.nio.file.attribute.FileAttribute...-

I am not saying that they should or should not be in the spec. Personally I have found them useful in the past, but I could live without them also.

**Comment by ChristianGruen** [comment]
Good to hear that there's some use for it, Adam. If there'll be a version 2 of the spec, I’m sure we'll keep it backward compatible.

**Comment by ChristianGruen** [comment]
Latest version of the function definition: https://qt4cg.org/specifications/expath-file-40/Overview.html#func-file-create-temp-file
